### PR TITLE
Move to version 3.x of the AWS Terraform provider

### PR DESCRIPTION
**Pull request by dav3r**
# 

## 🗣 Description
This PR moves this repo from using version 2 to version 3 of the [Terraform AWS provider](https://github.com/terraform-providers/terraform-provider-aws).

## 💭 Motivation and Context
This repo is used by [`cisagov/skeleton-ansible-role-with-test-user`](https://github.com/cisagov/skeleton-ansible-role-with-test-user), which needs to be updated to 3.x, so this repo must be updated first.

## 🧪 Testing
* I (temporarily) pointed this repo to the 3.x testing branch of [`cisagov/ci-iam-user-tf-module`](https://github.com/cisagov/ci-iam-user-tf-module) with https://github.com/cisagov/molecule-iam-user-tf-module/commit/35a496c9246699451f3e6e323c93a7bb327f06fe, then verified that I could successfully `terraform apply` and `terraform destroy` the [example code](https://github.com/cisagov/molecule-iam-user-tf-module/tree/develop/examples/basic_usage).  **Note that I will revert https://github.com/cisagov/molecule-iam-user-tf-module/commit/35a496c9246699451f3e6e323c93a7bb327f06fe after https://github.com/cisagov/ci-iam-user-tf-module/pull/14 is merged.**
* All pre-commit hooks pass.
## 🚥 Types of Changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (causes existing functionality to change)

## ✅ Checklist

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: user.tf [hunk 1]
```diff
@@ -1,5 +1,5 @@
 module "ci_user" {
-  source = "github.com/cisagov/ci-iam-user-tf-module"
+  source = "github.com/cisagov/ci-iam-user-tf-module?ref=improvement%2Fuse_tf_aws_3.x_provider"
```
Comment (jsf9k) [inline user.tf]: Don't forget to change this back!

**Review by jsf9k — APPROVED** [review]
"More weight."  - Giles Corey

**Review by hillaryj — APPROVED** [review]
🥉

File: user.tf (hunk 1 above)
Comment (dav3r) [reply user.tf]: Reverted with https://github.com/cisagov/molecule-iam-user-tf-module/pull/17/commits/e55e6efc85be9ad7684e5afc65b2a2e7fab0e5ce

**Pull request merged**